### DISCUSSION TITLE
Why is useLazyAsyncData still blocking my page load?

### DISCUSSION THREAD

**Discussion by NeonMika** [nuxt/nuxt#18090] [Questions] [answered]
The docs for data fetching state for `useLazyAsyncData `_This composable behaves identically to useAsyncData with the lazy: true option set. In other words, the async function does not block navigation_.

I have the following example that just sleeps 5 seconds within `useLazyAsyncData`. When I open http://localhost:3000/, nothing happens for 5 seconds.
I expected that the page will load immediately, and that the data will be set 5 seconds later.

I think that the [Data Fetching](https://v3.nuxtjs.org/docs/usage/data-fetching) documentation has to greatly be enhanced.
It is really hard to read for beginners, and it leaves open so many questions.

```
<template>
  <div>
    {{data ? data : -1}}
  </div>
</template>

<script setup>
console.log("before")
const { data } = await useLazyAsyncData('count', async () => {
  function sleep(ms) {
    return new Promise((resolve) => {
      setTimeout(resolve, ms);
    });
  }
  
  console.log("before load")
  await sleep(5000);
  console.log("after load")
  return 1
})
console.log("after")
</script>
```

**Comment by Walter-Melon** [comment, marked as answer]
It works when you set `ssr: false` inside `nuxt.config.js`.

```
import { defineNuxtConfig } from 'nuxt3';

// https://v3.nuxtjs.org/docs/directory-structure/nuxt.config
export default defineNuxtConfig({
  ssr: false,
});
```

**Reply by NeonMika** [reply]
Thank you @Walter-Melon!
Is this behavior describe anywhere?
Otherwise, this has be become a part of the docs - how should anybody new to Nuxt 3 guess this?
I also expected server-side rendered content to be non-blocking then. That is one reason why I use such a tightly coupled frontend-backend-framework such as Nuxt.

**Comment by danielroe** [comment]
If you don't want data fetching utilities to block the initial render, set `{ server: false }` in the options you pass. (See [documentation](https://v3.nuxtjs.org/docs/usage/data-fetching/).) If we are fetching on server-side, we _have_ to resolve it before  returning the HTML or the data fetching request will be orphaned on the server.

```vue

<script setup>
const { data } = await useLazyAsyncData('count', () => $fetch('/api/count'), { server: false })
</script>

<template>
  Page visits: {{ data }}
</template>
```
